### PR TITLE
Dependency cleanup

### PR DESCRIPTION
**Pull request by Ehbraheem**
- [x] Upgrade Hapi to v18
- [x] Upgrade Joi
- [x] Remove deprecated packages (good, good-console, etc).

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -55,19 +54,16 @@
     "typescript": "^3.8.2"
   },
   "dependencies": {
-    "@hapi/hapi": "17.9.0",
-    "@hapi/inert": "^6.0.1",
-    "@hapi/vision": "^6.0.0",
+    "@hapi/hapi": "18.4.1",
+    "@hapi/inert": "^5.2.2",
+    "@hapi/joi": "15.1.1",
+    "@hapi/vision": "^5.5.2",
```
Comment (jabdul) [inline package.json:60]: Install exact, remove `^`

File: src/users/routes.test.ts [hunk 2]
```diff
@@ -28,7 +28,7 @@ describe(`Routes: ${ROUTE_NAME}`, () => {
     const router: any = create({ services, validate }); // eslint-disable-line @typescript-eslint/no-explicit-any
     const responseData = 'Thanks for opening account Mr Abiodun Abdul!!!';
     const statusCode = 201;
-    const contentType = 'application/hal+json';
+    const contentType = 'application/json';
     let mockRequest = { log: null };
```
Comment (jabdul) [inline src/users/routes.test.ts:32]: Do we still need this here?

File: src/users/services.test.ts [hunk 3]
```diff
@@ -28,8 +26,8 @@ describe('Services: users', () => {
       };
       const log = jest.fn();
       const users = services(db).users;
-      await expect(users.create({ uuid, json, log })).rejects.toThrow();
-      await expect(users.create({ uuid, json, log })).rejects.toThrowError('false');
+      await expect(users.create({ log, payload: user })).rejects.toThrow();
+      await expect(users.create({ log, payload: user })).rejects.toThrowError('false');
```
Comment (jabdul) [inline src/users/services.test.ts:30]: Is the `log` still relevant here?

File: src/users/routes.test.ts (hunk 2 above)
Comment (Ehbraheem) [reply src/users/routes.test.ts]: Yes, we certainly do.

File: src/users/services.test.ts (hunk 3 above)
Comment (Ehbraheem) [reply src/users/services.test.ts]: Not really. Removed

File: package.json (hunk 1 above)
Comment (Ehbraheem) [reply package.json]: 👌

**Comment by Ehbraheem** [comment]
> You need to add usage of `pino` logger in the code.

I have done that now. There's a section for that also in the **README** with link to the documentation.

**Pull request merged**